# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.cpp
# Commit: cad764eb15a4
# Full Hash: cad764eb15a4a42cfdb2468ce762029130c0b9fc
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-30 10:00:42
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P12 top-level IPC stream filter should happen before OnStartRequest, r=mayhemer,necko-reviewers
#   
#   The WebRequest will suspend the channel in nsHttpChannel::PrepareToConnect(), apply the stream filter, send the main thread IPC  and resume.
#   That is, PHttpChannel::AttachStreamFilter should be handled in Child before OnStartRequest goes to listener.
#   
# ==============================================================================

diff -r f75cc7d5f40e -r cad764eb15a4 netwerk/protocol/http/HttpChannelChild.cpp
--- a/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 30 00:34:24 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 30 00:34:32 2020 +0000
@@ -194,7 +194,7 @@
       mIsLastPartOfMultiPart(false),
       mSuspendForWaitCompleteRedirectSetup(false),
       mRecvOnStartRequestSentCalled(false),
-      mSuspendedByWaitingForPermissionAndCookie(false) {
+      mSuspendedByWaitingForPermissionCookieStreamFilter(false) {
   LOG(("Creating HttpChannelChild @%p\n", this));
 
   mChannelCreationTime = PR_Now();
@@ -406,8 +406,8 @@
 
   mRecvOnStartRequestSentCalled = true;
 
-  if (mSuspendedByWaitingForPermissionAndCookie) {
-    mSuspendedByWaitingForPermissionAndCookie = false;
+  if (mSuspendedByWaitingForPermissionCookieStreamFilter) {
+    mSuspendedByWaitingForPermissionCookieStreamFilter = false;
     mEventQ->Resume();
   }
   return IPC_OK();
@@ -551,7 +551,7 @@
     MOZ_ASSERT(NS_IsMainThread());
 
     mEventQ->Suspend();
-    mSuspendedByWaitingForPermissionAndCookie = true;
+    mSuspendedByWaitingForPermissionCookieStreamFilter = true;
     mEventQ->PrependEvent(MakeUnique<NeckoTargetChannelFunctionEvent>(
         this, [self = UnsafePtr<HttpChannelChild>(this)]() {
           self->DoOnStartRequest(self, nullptr);